iotdb/src/main/java/org/apache/iotdb/db/query/aggregation/impl/MaxTimeAggrFunc.java

```diff
@@ -0,0 +1,146 @@
+import org.apache.iotdb.db.query.reader.IPointReader;
+import org.apache.iotdb.db.query.reader.merge.EngineReaderByTimeStamp;
+import org.apache.iotdb.db.utils.TimeValuePair;
+import org.apache.iotdb.db.utils.TsPrimitiveType;
+import org.apache.iotdb.tsfile.file.header.PageHeader;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.read.common.BatchData;
+
+public class MaxTimeAggrFunc extends AggregateFunction {
+
+  public MaxTimeAggrFunc() {
+    super(AggregationConstant.MAX_TIME, TSDataType.INT64);
+  }
+
+  @Override
+  public void init() {
+
+  }
+
+  @Override
+  public BatchData getResult() {
+    return resultData;
+  }
+
+  @Override
+  public void calculateValueFromPageHeader(PageHeader pageHeader) throws ProcessorException {
+    long maxTimestamp = pageHeader.getMaxTimestamp();
+
+    //has not set value
+    if (resultData.length() == 0) {
+      resultData.putTime(0);
+      resultData.putLong(maxTimestamp);
+      return;
+    }
+
+    if (resultData.getLong() < maxTimestamp) {
+      resultData.setLong(0, maxTimestamp);
+    }
+  }
+
+  @Override
+  public void calculateValueFromPageData(BatchData dataInThisPage, IPointReader unsequenceReader)
+      throws IOException, ProcessorException {
+    long time = -1;
+    int maxIndex = dataInThisPage.length() - 1;
+    if (maxIndex < 0) {
+      return;
+    }
+    time = dataInThisPage.getTimeByIndex(maxIndex);
+    while (unsequenceReader.hasNext()) {
+      if (unsequenceReader.current().getTimestamp() <= time) {
+        unsequenceReader.next();
+      } else {
+        break;
+      }
+    }
+    if (resultData.length() == 0) {
+      if (time != -1) {
+        resultData.putTime(0);
+        resultData.putAnObject(time);
+      }
+    } else {
+      //has set value
+      if (time != -1 && time > resultData.currentTime()) {
+        resultData.setAnObject(0, time);
+      }
+    }
+  }
+
+  @Override
+  public void calculateValueFromUnsequenceReader(IPointReader unsequenceReader)
+      throws IOException, ProcessorException {
+    TimeValuePair pair = null;
+    while (unsequenceReader.hasNext()) {
+      pair = unsequenceReader.next();
+    }
+    if (resultData.length() == 0) {
+      if (pair != null) {
+        resultData.putTime(0);
+        resultData.putAnObject(pair.getTimestamp());
+      }
+    } else {
+      //has set value
+      if (pair != null && pair.getTimestamp() > resultData.currentTime()) {
+        resultData.setAnObject(0, pair.getTimestamp());
+      }
+    }
+  }
+
+  @Override
+  public void calcAggregationUsingTimestamps(List<Long> timestamps,
+      EngineReaderByTimeStamp dataReader) throws IOException, ProcessorException {
+    long time = -1;
+    for (int i = 0; i < timestamps.size(); i++) {
+      TsPrimitiveType value = dataReader.getValueInTimestamp(timestamps.get(i));
+      if (value != null) {
+        time = timestamps.get(i);
+      }
+    }
```

Comment:
EngineReaderByTimeStamp is chronological. If just simply go through the timestamp list in reverse order, it will appear that when there is no corresponding value under the large timestamp, the value with small timestamp can not be read out because of excessive offset.  And because the data in a chunk is ordered and there is overlap between the chunks of the disordered data, it may be necessary to cache all the disordered data in order to achieve complete inversion traversal. This situation is complicated, so I added a todo for future optimization.